router/connection.go

```diff
@@ -292,7 +309,13 @@ func (conn *LocalConnection) run(actionChan <-chan ConnectionAction, finished ch
 		return
 	}
 
-	conn.Log("completed handshake; using protocol version", conn.version)
+	if conn.SessionKey == nil {
+		conn.Decryptor = NewNonDecryptor()
+	} else {
+		conn.Decryptor = NewNaClDecryptor(conn.SessionKey, conn.outbound)
+	}
```

Comment:
Unfortunately `conn.SessionKey` is required by `LocalConnection.ensureForwarders` for creating the encryptors. Which also suggests that having the decryptor initialisation in `LocalConnection` rather than in `ProtocolIntroResults.setup[No]Crypto` is the right thing.